### PR TITLE
feat(kudos): enable mentions in retro reflections

### PR DESCRIPTION
**Pull request by igorlesnenko**
Related to #9234. We need mentions in reflections to later use it for sending kudos

image

**How to test:**

- Start a retro
- Add few reflections and see that you can mention team users in them
- See that in all other places mention is rendered correctly — vote phase, summary etc.
- Because of some refactoring, play with retro and make sure nothing's broken in general for reflection cards or and reflections editor

### CODE REVIEW

File: packages/client/components/ReflectionGroup/RemoteReflection.tsx [hunk 1]
```diff
@@ -273,6 +275,16 @@ const RemoteReflection = (props: Props) => {
             editorState={editorState}
             readOnly
             disableAnonymity={disableAnonymity}
+            teamId={teamId}
+            editorRef={editorRef}
+            placeholder=''
+            setEditorState={() => {
```
Comment (nickoferrall) [inline packages/client/components/ReflectionGroup/RemoteReflection.tsx:281]: +1 could this and `handleKeyDownFallback` and `placeholder` be optional props in `ReflectionEditorWrapper`?

**Review by nickoferrall — APPROVED** [review]

File: packages/client/components/RetroReflectPhase/PhaseItemEditor.tsx [hunk 2]
```diff
@@ -272,6 +283,9 @@ const PhaseItemEditor = (props: Props) => {
                   editorState={card.editorState}
                   readOnly
                   disableAnonymity={disableAnonymity}
+                  teamId={teamId}
+                  editorRef={editorRef}
```
Comment (Dschoordsch) [inline packages/client/components/RetroReflectPhase/PhaseItemEditor.tsx:287]: -1 this part seems wrong. If we have some cards in flight, they should not need a reference to the editor anymore as I should not be able to do any edits anymore

**Review by Dschoordsch — CHANGES_REQUESTED** [review]
Overall looks good. The added functions, like mentions and links work. However there is something weird happening with the editor state: https://www.loom.com/share/4307debd45fb49968183f7c68f58ca94?sid=9cdf50f6-16dc-4d9c-a211-c46e7328e49a

**Review by Dschoordsch — APPROVED** [review]
Works.
One thing that's missing: we're not sending any notifications for a mention in a reflection. Will this be part of the next step?

**Comment by igorlesnenko** [comment]
> One thing that's missing: we're not sending any notifications for a mention in a reflection. Will this be part of the next step?

Going to work on it as a next issue: https://github.com/ParabolInc/parabol/issues/9328

**Pull request merged**